### ISSUE TITLE
Unable to boot Rasperry Pi 3B+ with bcmrpi3-kernel-4.19.69.20190903

### ISSUE THREAD

**Issue by davinellulinvega** [sakaki-/bcmrpi3-kernel#10]
Hello all,

So I have been running Arch Linux ARM (the aarch64 version) for a long time on my raspberry pi 3B+. All was well with the world, until I updated the `linux-aarch64-raspberrypi-bin` package. The update went well, no error message. I went to reboot the RPI and ... **nothing**.
When I say **nothing**, what I mean is that the red led is ON, so power is good. The green (act) led flashes in a fixed pattern of 7 flashes, followed by a pause, then again 7 flashes. According to [this lovely website](https://elinux.org/R-Pi_Troubleshooting#Green_LED_blinks_in_a_specific_pattern), this seems to indicate that the `kernel.img` file was not found. If I plug a screen via HDMI it stays on the rainbow and nothing else appear, no messages no nothing.
I checked both file systems and SD card. Everything seem fine. So now I am a bit at a loss as to what I should do/try next? Can I simply copy a kernel image from a previous release, paste it in the `/boot` folder and call it a day? Or should I just re-install everything? Finally, am I the only one in a pickle here? Or are others experiencing the same issue?
Thank you very much in advance for any help you can provide.

Sincerely.

**Comment by sakaki-** [comment]
Hi @davinellulinvega,

Not sure what the issue is here; it is entirely possible there is an issue with the specific config or build of course, since these kernels are auto-created. The last variant I tested explicitly was 4.19.67.20190827, as that was used in the v1.5.1 release of [gentoo-on-rpi-64bit](https://github.com/sakaki-/gentoo-on-rpi-64bit) (you can see the tested variants listed [here](https://github.com/sakaki-/genpi64-overlay/blob/master/profiles/targets/rpi3/package.unmask/bcmrpi3-kernel-bin)).

You do need to make sure you are using boot firmware of the same currency as your kernel.

Please also note that lifof reports [here](https://github.com/sakaki-/bcmrpi3-kernel/issues/9#issuecomment-528864516) that 4.19.69.20190903 was working for them.

Best, sakaki

**Comment by davinellulinvega** [comment]
Hello @sakaki-,

Thank you very much for taking the time to answer my question. Even though I am not completely new to the Linux world, I have to admit that I have never meddled with the kernel or the boot process for that matters. As a consequence, when you say:

> You do need to make sure you are using boot firmware of the same currency as your kernel.

you might be right, but I have no idea what you are talking about.
As for the other issue you referenced, it gives me the impression that I should try booting my RPI 3B+ with an empty `config.txt` file and see what happens.
Once again, thank you very much for your help.

Sincerely.

**Comment by sakaki-** [comment]
Hi @davinellulinvega, 

By "[boot firmware](https://www.raspberrypi.org/documentation/configuration/boot_folder.md)", I mean (on an RPi3) the closed-source `/boot/bootcode.bin`, `/boot/fixup*.dat` and `/boot/start*.elf` files, which are used in the early-stage startup of the device (the `/boot/overlays` directory comes with the kernel package, as do the top-level `/boot/*.dtb` files, so you don't need to worry about them; the `/boot/cmdline.txt` and `/boot/config.txt` files you presumably manage yourself).

Said boot firmware files have been undergoing significant revisions lately. On my Gentoo image, these are kept in revision lockstep with the kernel, but that may not be so in Arch.

If in doubt, install a copy of vanilla Raspbian Buster on a separate microSD card, run `sudo rpi-update` to update this firmware, and then copy it across to the `/boot` on your target image.

Best, sakaki

**Comment by davinellulinvega** [comment]
Hello @sakaki-,

Thank you very much for the detailed explanation.
I did some more research in the meantime and it seems that your suspicions are correct.
To give you the complete story. I also have an Odroid-XU4, running arch linux. Funnily enough, at the same time the RPI 3B+ went down, so did the Odroid. After some quick search, I found that the kernel for all ARM images had changed (I sincerely do not remember where I got this info) and that some people where having trouble with their board hanging on reboot. That is what lead me to check what the last upgraded package was on both of my computers. Well, without too much surprise it was indeed a kernel update. As a consequence I posted this issue here to know if more people were in trouble and if there was an *easy* solution to this problem. After more research on the Odroid forum, I stumbled across [this thread](https://forum.odroid.com/viewtopic.php?t=36234). The admin/dev there provided with a `zImage` that should apparently fix the issue. However, in my case simply replacing the `zImage` did not do the trick. I think the reason why it did not work is as you have explained. I will have to replace and match not only the kernel, but also some of the other files in the `/boot` folder.
Since I do not have much installed on the RPI and Odroid yet, and I prefer not to mess with things I do not understand, I will make a clean install of both systems and make sure to back them up before any upcoming upgrade.
Thank you very much for taking the time to answer my question and keep up the awesome work.

Sincerely.
